### ISSUE TITLE
Save Actions plugin not working when Eclipse Code Formatter plugin is active

### ISSUE THREAD

**Issue by isidorakandic** [dubreuia/intellij-plugin-save-actions#330]
Before a recent update I was able to use both plugins at the same time. After the update, I need to switch code formatter from Eclipse to default in order to apply Save Actions plugin. When code formatter is set to Eclipse, clicking on Save Actions shortcut has no effect.

I'm currently using IntelliJ Idea Ultimate 2020.2, but the problem exists with 2020.1.3 as well, maybe a few releases before that.

**Comment by krasa** [comment]
Maybe I need to add this back? https://github.com/krasa/EclipseCodeFormatter/blob/afee8a8cb857d04e45bda9e712a142663fa2398d/src/java/krasa/formatter/plugin/EclipseCodeFormatter.java#L97-L106

https://github.com/krasa/EclipseCodeFormatter/issues/214


I found this in the log:
```
2020-08-06 20:47:15,272 [ 289148]  ERROR - re.component.SaveActionManager - Attempt to modify PSI for non-committed Document! 
java.lang.IllegalStateException: Attempt to modify PSI for non-committed Document!
	at com.intellij.pom.core.impl.PomModelImpl.startTransaction(PomModelImpl.java:316)
	at com.intellij.pom.core.impl.PomModelImpl.lambda$runTransaction$2(PomModelImpl.java:135)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:629)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:581)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:192)
	at com.intellij.pom.core.impl.PomModelImpl.runTransaction(PomModelImpl.java:133)
	at com.intellij.psi.impl.source.tree.ChangeUtil.prepareAndRunChangeAction(ChangeUtil.java:143)
	at com.intellij.psi.impl.source.tree.CompositeElement.addChild(CompositeElement.java:579)
	at com.intellij.psi.impl.source.tree.CompositeElement.addChildren(CompositeElement.java:673)
	at com.intellij.psi.impl.source.codeStyle.CodeEditUtil.addChildren(CodeEditUtil.java:53)
	at com.intellij.psi.impl.source.tree.CompositeElement.addInternal(CompositeElement.java:447)
	at com.intellij.psi.impl.source.tree.java.ModifierListElement.addInternal(ModifierListElement.java:47)
	at com.intellij.psi.impl.source.PsiModifierListImpl.setModifierProperty(PsiModifierListImpl.java:253)
	at com.intellij.psi.util.PsiUtil.setModifierProperty(PsiUtil.java:1146)
	at com.intellij.codeInspection.localCanBeFinal.LocalCanBeFinal$AcceptSuggested.applyFix(LocalCanBeFinal.java:327)
	at com.intellij.codeInspection.localCanBeFinal.LocalCanBeFinal$AcceptSuggested.applyFix(LocalCanBeFinal.java:313)
	at com.dubreuia.processors.java.InspectionRunnable.writeQuickFixes(InspectionRunnable.java:71)
	at com.dubreuia.processors.java.InspectionRunnable.run(InspectionRunnable.java:46)
	at com.dubreuia.processors.SaveWriteCommand$1.run(SaveWriteCommand.java:32)
	at com.intellij.openapi.application.RunResult.run(RunResult.java:35)
	at com.intellij.openapi.command.WriteCommandAction.lambda$performWriteCommandAction$1(WriteCommandAction.java:246)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:980)
	at com.intellij.openapi.command.WriteCommandAction.lambda$performWriteCommandAction$2(WriteCommandAction.java:245)
	at com.intellij.openapi.command.WriteCommandAction.lambda$doExecuteCommand$4(WriteCommandAction.java:303)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:220)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:187)
	at com.intellij.openapi.command.WriteCommandAction.doExecuteCommand(WriteCommandAction.java:305)
	at com.intellij.openapi.command.WriteCommandAction.performWriteCommandAction(WriteCommandAction.java:244)
	at com.intellij.openapi.command.WriteCommandAction.execute(WriteCommandAction.java:225)
	at com.dubreuia.processors.SaveWriteCommand.execute(SaveWriteCommand.java:35)
	at com.dubreuia.core.component.Engine.lambda$processPsiFiles$5(Engine.java:81)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$11$1.accept(ReferencePipeline.java:442)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at com.dubreuia.core.component.Engine.processPsiFiles(Engine.java:82)
	at com.dubreuia.core.component.Engine.processPsiFilesIfNecessary(Engine.java:65)
	at com.dubreuia.core.component.SaveActionManager.guardedProcessPsiFiles(SaveActionManager.java:144)
	at com.dubreuia.core.component.SaveActionManager.lambda$beforeDocumentsSaving$3(SaveActionManager.java:128)
	at java.base/java.util.HashMap.forEach(HashMap.java:1336)
	at com.dubreuia.core.component.SaveActionManager.beforeDocumentsSaving(SaveActionManager.java:127)
	at com.dubreuia.core.component.SaveActionManager.beforeAllDocumentsSaving(SaveActionManager.java:112)
	at jdk.internal.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.intellij.util.messages.impl.MessageBusImpl.invokeListener(MessageBusImpl.java:632)
	at com.intellij.util.messages.impl.MessageBusImpl.deliverMessage(MessageBusImpl.java:406)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:384)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:366)
	at com.intellij.util.messages.impl.MessageBusImpl.access$100(MessageBusImpl.java:32)
	at com.intellij.util.messages.impl.MessageBusImpl$MessagePublisher.invoke(MessageBusImpl.java:187)
	at com.sun.proxy.$Proxy31.beforeAllDocumentsSaving(Unknown Source)
	at jdk.internal.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.multiCast(FileDocumentManagerImpl.java:174)
	at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.lambda$new$0(FileDocumentManagerImpl.java:129)
	at com.sun.proxy.$Proxy31.beforeAllDocumentsSaving(Unknown Source)
	at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.saveAllDocuments(FileDocumentManagerImpl.java:323)
	at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.saveAllDocuments(FileDocumentManagerImpl.java:313)
	at com.intellij.ide.actions.SaveAllAction.actionPerformed(SaveAllAction.kt:20)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:282)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$1.performAction(IdeKeyEventDispatcher.java:610)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.lambda$processAction$3(IdeKeyEventDispatcher.java:670)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:94)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:669)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:620)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processActionOrWaitSecondStroke(IdeKeyEventDispatcher.java:517)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.inInitState(IdeKeyEventDispatcher.java:471)
	at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:220)
	at com.intellij.ide.IdeEventQueue.dispatchKeyEvent(IdeEventQueue.java:888)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:833)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:450)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:744)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:449)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:802)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:503)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
2020-08-06 20:47:15,278 [ 289154]  ERROR - re.component.SaveActionManager - IntelliJ IDEA 2020.2  Build #IU-202.6397.94 
2020-08-06 20:47:15,278 [ 289154]  ERROR - re.component.SaveActionManager - JDK: 11.0.3; VM: Java HotSpot(TM) 64-Bit Server VM; Vendor: Oracle Corporation 
2020-08-06 20:47:15,279 [ 289155]  ERROR - re.component.SaveActionManager - OS: Windows 10 
2020-08-06 20:47:15,279 [ 289155]  ERROR - re.component.SaveActionManager - Last Action: SaveAll 
2020-08-06 20:47:15,279 [ 289155]  ERROR - re.component.SaveActionManager - Current Command: Undefined 
```

**Comment by krasa** [comment]
Fixed in Eclipse Code Formatter plugin 20.3.201.6668.0-Eclipse_2020-06-JavaOnly

**Comment by dubreuia** [comment]
Great, thanks @krasa
